src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/AutoInsert/OnAutoInsertEndpoint.cs

```diff
@@ -53,17 +52,17 @@ internal class OnAutoInsertEndpoint(
 
     public void ApplyCapabilities(VSInternalServerCapabilities serverCapabilities, VSInternalClientCapabilities clientCapabilities)
     {
-        var triggerCharacters = _onAutoInsertProviders.Select(provider => provider.TriggerCharacter);
+        var triggerCharacters = _autoInsertService.TriggerCharacters;
 
         if (_languageServerFeatureOptions.SingleServerSupport)
         {
-            triggerCharacters = triggerCharacters.Concat(s_htmlAllowedTriggerCharacters).Concat(s_cSharpAllowedTriggerCharacters);
+            triggerCharacters = triggerCharacters
+                .Concat(AutoInsertService.HtmlAllowedAutoInsertTriggerCharacters)
+                .Concat(AutoInsertService.CSharpAllowedAutoInsertTriggerCharacters)
+                .ToFrozenSet(StringComparer.Ordinal);
         }
 
-        serverCapabilities.OnAutoInsertProvider = new VSInternalDocumentOnAutoInsertOptions()
-        {
-            TriggerCharacters = triggerCharacters.Distinct().ToArray()
-        };
+        serverCapabilities.EnableOnAutoInsert(triggerCharacters);
```

Comment:
I switched it to ImmutableList, not sure if that causes the same issue though?

Reply:
Consider using `ImmutableArray<T>` instead of `ImmutableList<T>`. That's generally what we use. In fact, if you search the repo for "ImmutableList", you'll find that this would be the _first_ usage of `ImmutableList<T>`. So, if there's a good reason to prefer `ImmutabeList<T>` over `ImmutableArray<T>`, you'll probably want to add a comment to document the reason.

Reply:
Sorry, I meant ImmutableArray<T> even though that's clearly not what I typed :) It is ImmutableArray<T> already. Thank you!